src/lib/UniversalDiscovery/Event/ConfigResolveEvent.php

```diff
@@ -21,6 +21,7 @@ class ConfigResolveEvent extends Event
         'subtree_search',
         'single',
         'multiple',
+        'single_container',
```

Comment:
I don't think we should enforce read-only on `single_container` configuration (and we shouldn't enforce it on `single` and `multiple` either). If a feature requires specific treatment it should us own configuration. Now it is global change and affects all places where `single_container` is used.

If there is an issue with move/swap it means they should use own configuration (will it be copy of `single_container` or not) and that configuration could be read-only.

The same applies for the `single` and `multiple` above. If there was an issue with relations, relations should use own configuration instead of global ones (and we shouldn't make global ones read-only).